plugin/HandlerSocket-Plugin-for-MySQL/libhsclient/auto_ptrcontainer.hpp

```diff
@@ -56,7 +54,12 @@ struct auto_ptrcontainer {
   }
   reference operator [](size_type n) { return cnt[n]; }
   const_reference operator [](size_type n) const { return cnt[n]; }
-  void clear() { cnt.clear(); }
+  void clear() {
+    for (iterator i = begin(); i != end(); i++) {
+      delete *i;
+    }
+    cnt.clear();
+  }
```

Comment:
For containers like this hiding copy constructor and assignment operator is also a good idea.